client/battle/CCreatureAnimation.cpp

```diff
@@ -210,20 +210,26 @@ bool CCreatureAnimation::incrementFrame(float timePassed)
 {
 	elapsedTime += timePassed;
 	currentFrame += timePassed * speed;
+	auto framesNumber = framesInGroup(type);
+	bool ret = false;
 
-	if (currentFrame >= float(framesInGroup(type)))
+	if(framesNumber <= 0)
+	{
+		endAnimation();
+	}
+	else if(currentFrame >= float(framesNumber))
 	{
 		// just in case of extremely low fps (or insanely high speed)
-		while (currentFrame >= float(framesInGroup(type)))
-			currentFrame -= framesInGroup(type);
+		while(currentFrame >= float(framesNumber))
+			currentFrame -= framesNumber;
 
-		if (once)
+		if(once)
 			setType(CCreatureAnim::HOLDING);
 
 		endAnimation();
-		return true;
+		ret = true;
 	}
-	return false;
+	return ret;

```

Comment:
Do you prefer a single return in a method? I would leave it as is.

Reply:
Ok

Reply:
Ok